tree-sitter-highlight.el

```diff
@@ -220,8 +237,19 @@ to faces.  Each function takes no arguments."
                           (ts-byte-from-position start)
                           (ts-byte-from-position end)))))
 
+(defun tree-sitter-highlight--maybe-run-idle-timer ()
+  (unless tree-sitter-highlight--idle-timer-scheduled
+    (setq tree-sitter-highlight--idle-timer-scheduled t)
+    (run-with-idle-timer tree-sitter-highlight-idle-timer-delay nil
+                         (lambda ()
+                           (setq tree-sitter-highlight--idle-timer-scheduled nil)
+                           (tree-sitter-highlight--highlight nil (point-min) (point-max))))))
+
+
 (defun tree-sitter-highlight--after-change (old-tree)
-  (tree-sitter-highlight--highlight old-tree (window-start) (window-end)))
+  (tree-sitter-highlight--highlight old-tree (window-start) (window-end))
```

Comment:
Just curious, have you considered providing tree-sitter implementations for font-lock-fontify-region-function(and related) ? Thus you will be able to share the logic related to calculating what has to be highlighted.

Reply:
I think if we would want to do that we would want to choose `jit-lock` instead, `font-lock` relies on that (afaik) for knowing when to fontify what.
I've considered it, but didn't implement it yet, since this way is easier. I can add it as a toggle maybe, to see if it would improve things (mainly that currently I can either delay highlighting in case of errors, or eagerly highlight but have to clear (remove all faces from) the whole window first, which causes the part after the error to not be highlighted (which you technically can't do correctly because there's an error after all) and which also causes "flickering" if you type fast enough to were you rapidly produce/remove errors in the syntax tree). However I think it's unlikely that `jit-lock` will help with that specific problem, because we already know what changed, and if i only clear/highlight what changed, then sometimes things are not highlighted at all.

Reply:
Update: I've tried both `font-lock` and `jit-lock`. `font-lock` seems to work better, so I've only kept the code for that. Thanks for the suggestion.